src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -135,55 +137,163 @@ void CommissionerDiscoveryController::Ok()
     }
     client->SetUDCClientProcessingState(UDCClientProcessingState::kObtainingOnboardingPayload);
 
-    if (mPasscodeService != nullptr)
+    if (mPasscodeService == nullptr)
     {
-        char rotatingIdString[chip::Dnssd::kMaxRotatingIdLen * 2 + 1] = "";
-        Encoding::BytesToUppercaseHexString(client->GetRotatingId(), client->GetRotatingIdLength(), rotatingIdString,
-                                            sizeof(rotatingIdString));
-        // Encoding::BytesToUppercaseHexString(client->GetRotatingId(), chip::Dnssd::kMaxRotatingIdLen, rotatingIdString,
-        //                                     sizeof(rotatingIdString));
+        HandleContentAppPasscodeResponse(0);
+        return;
+    }
+
+    char rotatingIdString[chip::Dnssd::kMaxRotatingIdLen * 2 + 1] = "";
+    Encoding::BytesToUppercaseHexString(client->GetRotatingId(), client->GetRotatingIdLength(), rotatingIdString,
+                                        sizeof(rotatingIdString));
+    CharSpan rotatingIdSpan = chip::CharSpan(rotatingIdString, sizeof(rotatingIdString));
```

Comment:
This size looks wrong.

(And separately, why the extra chip:: prefix?)

Reply:
removed the prefix. What about the size does not look right?

Reply:
Well, what will be contained in the span given the size that's being used?

Reply:
fixed

Reply:
I missed that the size comment was referring to the CharSpan. It's true that we don't need the entire max length, we only need the portion through the null termination. Fixed.